Source/WebKit/UIProcess/API/C/WKWebsiteDataStoreRef.cpp

```diff
@@ -879,9 +879,26 @@ void WKWebsiteDataStoreResetQuota(WKWebsiteDataStoreRef dataStoreRef, void* cont
     });
 }
 
+void WKWebsiteDataStoreResetStoragePersistedState(WKWebsiteDataStoreRef dataStoreRef, void* context, WKWebsiteDataStoreResetStoragePersistedStateCallback callback)
+{
+    WebKit::toImpl(dataStoreRef)->resetStoragePersistedState([context, callback] {
+        if (callback)
+            callback(context);
+    });
+}
+
 void WKWebsiteDataStoreClearStorage(WKWebsiteDataStoreRef dataStoreRef, void* context, WKWebsiteDataStoreClearStorageCallback callback)
 {
-    WebKit::toImpl(dataStoreRef)->clearStorage([context, callback] {
+    OptionSet<WebKit::WebsiteDataType> dataTypes = {
+        WebKit::WebsiteDataType::LocalStorage,
+        WebKit::WebsiteDataType::IndexedDBDatabases,
+        WebKit::WebsiteDataType::FileSystem,
```

Comment:
Maybe this is the key to the change? That we now use removeData instead of clearStorage, and clearStorage didn't clear storage for real?

Reply:
Yes